Automattic/woocommerce-payments
# Update to use wcpaySettings across pages to reduce duplication

Fixes #1482 

#### Changes proposed in this Pull Request

At present, the Account Overview page uses the javascript object _wcpaySettings_ with information such as accountFees, accountStatus, zeroDecimalCurrencies etc. The Settings page uses most of the same values, but these are again set into two javascript objects,  _wcpaySettings_ and _wcpayAdminSettings_ 

This PR
- creates a common $wcpay_settings to avoid duplication. This is then set as _wcpaySettings_ on both Account Overview and Settings pages.
- Removes _wcpayAdminSettings_.
- Passes only the required parameters from _wcpaySettings_ to each component.

The change is implemented as a refactoring within the register_payments_scripts() function.
There are no new tests affecting this refactoring, all existing tests passed.

#### Testing instructions

1) Switch to the branch.
2) On Admin Dashboard, go to WooCommerce > Settings > Payments, check that account status, base fee and other information is displayed.
3) Set wp_options '_wcpay_feature_account_overview' to true, on Admin Dashboard, go to Payments > Overview and check that base fee and other details are displayed.

-------------------
